common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java

```diff
@@ -247,22 +249,27 @@ public String getCurrentActiveBrokerPackageName() {
     public String getValidActiveBrokerPackageName() throws ClientException {
         final String methodTag = TAG + ":getValidActiveBrokerPackageName";
 
-        final int numberOfAuthenticators;
+        final AuthenticatorDescription[] authenticators;
         try {
-            final AuthenticatorDescription[] authenticators = AccountManager.get(mContext).getAuthenticatorTypes();
-            numberOfAuthenticators = authenticators.length;
-            Logger.info(methodTag, numberOfAuthenticators + " Authenticators registered.");
-            for (final AuthenticatorDescription authenticator : authenticators) {
-                if (BROKER_ACCOUNT_TYPE.equals(authenticator.type)) {
-                    verifySignatureAndThrow(authenticator.packageName);
-                    return authenticator.packageName;
-                }
-            }
+            authenticators = AccountManager.get(mContext).getAuthenticatorTypes();
         } catch (final Exception exception) {
-            final ClientException clientException = new ClientException(NOT_VALID_BROKER_FOUND, exception.getMessage());
+            final ClientException clientException = new ClientException(ACCOUNT_MANAGER_FAILED, exception.getMessage());
```

Comment:
So this is Account Manager being completely broken?

Reply:
this is account manager just failing unexpectedly